packages/fiori/src/Timeline.ts

```diff
@@ -105,51 +117,103 @@ class Timeline extends UI5Element {
 	}
 
 	_onfocusin(e: FocusEvent) {
-		const target = e.target as TimelineItem;
+		let target = e.target as ITimelineItem | ToggleButton;
+
+		if ((target as ITimelineItem).isGroupItem) {
+			target = target.shadowRoot!.querySelector<ToggleButton>("ui5-toggle-button")!;
+		}
 
 		this._itemNavigation.setCurrentItem(target);
 	}
 
 	onBeforeRendering() {
 		this._itemNavigation._navigationMode = this.layout === TimelineLayout.Horizontal ? NavigationMode.Horizontal : NavigationMode.Vertical;
 
+		if (this.items) {
+			for (let i = 0; i < this.items.length; i++) {
+				this.items[i].layout = this.layout;
+				if (this.items[i + 1] && !!this.items[i + 1].icon) {
+					this.items[i].forcedLineWidth = SHORT_LINE_WIDTH;
+				} else if (this.items[i].icon && this.items[i + 1] && !this.items[i + 1].icon) {
+					this.items[i].forcedLineWidth = LARGE_LINE_WIDTH;
+				}
+			}
+		}
+		this._setLastItem();
```

Comment:
Can these two method calls be moved before the check for items, and accordingly, can we reverse the check?
if (this.items) {
    return;
}

for........